prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,341 @@
+            settings.map{ it.getName() }
+        }
+    }
+
+    /**
+     * Validates the JWT Token supplied in the authorization header
+     * 
+     * @param request the HTTPRequest object
+     * @param logger logger
+     * 
+     * @returns user from the token; otherwise null
+     */
+    fun validateUser( request: HttpRequestMessage<String?>, logger: Logger): String? {
+        var jwtToken = request.headers["authorization"] ?: ""
+
+        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""
+
+        var user: String? = null;
+
+        if (jwtToken.isNotBlank()) {
+            try {
+                // get the access token verifier
+                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                    .setIssuer(OKTA_ISSUER)
+                    .build()
+                // get it to decode the token from the header
+                val jwt = jwtVerifier.decode(jwtToken)
+                    ?: throw Throwable("Error in validation of jwt token")
+                // get the user name
+                user = jwt.claims["sub"].toString();
+            }
+            catch (ex: Throwable) {
+                logger.log(Level.WARNING, "Error in verification of token", ex)
+            }
+        }
+        return user;
+    }
+
+
+    /**
+     * Converts an organization name to an OKTA group name
+     * 
+     * @param org organization name (ex. pima-az-phd )
+     * 
+     * @returns encoded org name (ex. DHpima_az_phd )
+     */
+    private fun encodeOrg( org: String ): String {
+        return "DH"+org.replace( "-","_" )
+    }
+
+    /**
+     * Retrieve a list of emails within an organization
+     * 
+     * @params org The organization to fetch emails for
+     * 
+     * @returns List of emails to send to
+     */
+    private fun getEmails( org: String, logger: Logger): List<String> {
+
+        var emails :MutableList<String> = mutableListOf()
+    
+        try{
+
+            var ssws: String? = SecretHelper.getSecretService().fetchSecret("SSWS_OKTA")
+
+            if( ssws !== null ){
+
+                var grp = encodeOrg( org );
+
+                // get the OKTA Group Id
+                var response1 = httpGet( url="${OKTA_GROUPS_API}?q=${grp}", 
+                                    headers=mapOf( "Authorization" to "SSWS ${ssws}" ) );
+                var grpId = ((response1.jsonArray).get( 0 ) as JSONObject).getString("id");
+
+                // get the users within that OKTA group
+                var response = httpGet( url="${OKTA_GROUPS_API}/${grpId}/users", 
+                                    headers=mapOf( "Authorization" to "SSWS ${ssws}" ) );
+            
+                for( user in response.jsonArray )
+                    emails.add( (user as JSONObject).getJSONObject("profile").getString("email") );
+            }
+        }
+        catch( ex: Throwable ){
+            logger.warning( "Error in fetching emails" )
```

Comment:
To make our future lives easier, can you include some information in this log entry as to why the e-mails could not be fetched (while still not including any private information)?
The exception could be due to a variety of reasons and dependencies. It could be due to something Okta related, it could be due to something network related, but it could also be due to something related to a user not being able to be parsed as a JSON object. Having that information in the warning will be useful when debugging.